### ISSUE TITLE
Question: Removal of lock_.EnterWriteLock();

### ISSUE THREAD

**Issue by LiorBanai** [reyntjesr/Hdf5DotnetTools#13]
Hi,
there are some places in code that are commented out regarding locks like:
`       
 public Hdf5AcquisitionFileWriter(string aFilename, string groupName = "/EEG")
        {
            H5E.set_auto(H5E.DEFAULT, null, IntPtr.Zero);
            //lock_.EnterWriteLock();
            _filename = aFilename;
            fileId = Hdf5.CreateFile(aFilename);
            _groupName = groupName;
            _groupId = Hdf5.CreateGroup(fileId, _groupName);

            Header = new Hdf5AcquisitionFile();
            _nrOfRecords = 0;
            _sampleCount = 0;
            //lock_.ExitWriteLock();
         
        }
`

and reason for the removal of it?

Thanks

**Comment by reyntjesr** [comment]
Hi,

When I look in the file history I see it was always commented out. Probably in an older version something went wrong while writing large data acquisition files and I tried this locking mechanism, but never really used it.

You removed the post build events in the project files that copy the bin32 and bin64 folders to the binary directories after a build.
Was there a reason for this? 

Again thanks for going through my code and correcting so many things!

Robert

**Comment by LiorBanai** [comment]
I have replaced the direct reference to the HDF.PInvoke dll with the Nuget package HDF.PInvoke.NETStandard.
replaced:
    <Reference Include="HDF.PInvoke, Version=1.10.5.1, Culture=neutral, PublicKeyToken=18793c1fcffe58a3, processorArchitecture=MSIL">
      <HintPath>..\packages\HDF.PInvoke.1.10.5.1\lib\HDF.PInvoke.dll</HintPath>
    </Reference>
with

    <PackageReference Include="HDF.PInvoke">
      <Version>1.10.5.2</Version>
    </PackageReference>


so during build the correct dll is being copied to the output folder so this is kind of redundant. Also when creating Nuget packages, the other depended packages will be added automatically.

I may also create nuget package for your code and I'll credit you of course. I'm still learning your code and I saw that in the class AttributeClass not all attributes are writing to the H5 file.
for example :
  [Hdf5Attribute("birthdate")]
   public DateTime aDatetime = new DateTime(1969, 12, 01, 12, 00, 00, DateTimeKind.Local);

the birthdate is not shown so as an exercise I'll try to see if I can add it..
